tests/components/sonos/test_init.py

```diff
@@ -132,3 +140,249 @@ async def test_async_poll_manual_hosts_warnings(
+        assert mock_async_call_later.call_count == 5
+
+
+async def patch_gethostbyname(host: str) -> str:
+    """Mock to return host name as ip address for testing."""
+    return host
+
+
+async def test_async_poll_manual_hosts_ping_failure(
+    hass: HomeAssistant, caplog: pytest.LogCaptureFixture
+) -> None:
+    """Test that a failure to ping device is handled properly."""
+    await async_setup_component(
+        hass,
+        sonos.DOMAIN,
+        {"sonos": {"media_player": {"interface_addr": "127.0.0.1"}}},
+    )
+    await hass.async_block_till_done()
+    manager: SonosDiscoveryManager = hass.data[DATA_SONOS_DISCOVERY_MANAGER]
+    with patch.object(manager, "_async_gethostbyname", side_effect=patch_gethostbyname):
+        manager.hosts.add("10.10.10.10")
+
+        with caplog.at_level(logging.DEBUG), patch.object(
+            manager, "_async_handle_discovery_message", new=AsyncMock()
+        ) as mock_discovery_message, patch(
+            "homeassistant.components.sonos.async_call_later"
+        ) as mock_async_call_later, patch(
+            "homeassistant.components.sonos.async_dispatcher_send"
+        ), patch.object(
+            hass, "async_add_executor_job", new=AsyncMock()
+        ) as mock_async_add_executor_job:
+            mock_async_add_executor_job.return_value = []
+            caplog.clear()
+
+            mock_discovery_message.side_effect = asyncio.TimeoutError("TimeoutError")
+            await manager.async_poll_manual_hosts()
+            assert len(caplog.messages) == 1
+            record = caplog.records[0]
+            assert record.levelname == "WARNING"
+            assert "Discovery message failed" in record.message
+            assert "TimeoutError" in record.message
+            mock_async_call_later.assert_called_once()
+
+
+async def test_async_poll_manual_hosts(hass: HomeAssistant) -> None:
+    """Tests the logic and execution branches for the function."""
+    await async_setup_component(
+        hass,
+        sonos.DOMAIN,
+        {"sonos": {"media_player": {"interface_addr": "127.0.0.1"}}},
+    )
+    await hass.async_block_till_done()
+    manager: SonosDiscoveryManager = hass.data[DATA_SONOS_DISCOVERY_MANAGER]
+
+    with patch.object(manager, "_async_gethostbyname", side_effect=patch_gethostbyname):
+        manager.hosts.add("10.10.10.1")
+        manager.hosts.add("10.10.10.2")
+
+        # Test 1 first device fails, second device successful, speakers do not exist
+        with patch.object(
+            manager, "_async_handle_discovery_message", new=AsyncMock()
+        ) as mock_discovery_message, patch(
+            "homeassistant.components.sonos.async_call_later"
+        ) as mock_async_call_later, patch(
+            "homeassistant.components.sonos.async_dispatcher_send"
+        ), patch.object(
+            hass,
+            "async_add_executor_job",
+            new=AsyncMock(),
+        ) as mock_async_add_executor_job:
```

Comment:
We shouldn't patch hass internals in integration tests

Reply:
Ok I'll rework tests and/or make helper functions in the class for these calls.  It'll improve readability of the test also.

Reply:
@bdraco - updated the test to remove patching of the hass object, please review.